actors/paych/src/lib.rs

```diff
@@ -147,7 +147,7 @@ impl Actor {
                 sv.channel_addr
             )
         })?;
-        if pch_addr != svpch_id_addr {
+        if pch_addr != Address::new_id(svpch_id_addr) {
```

Comment:
An alternative here, where we know we have an ID address, is 
`pch_addr.id().unwrap() == svpch_id`

The unwrap is a bit unpleasant and kinda points to us later changing `message.receiver()` to return the actor ID too. But it's probably better than wrapping all the time. I could be convinced otherwise. Any thoughts @ZenGround0 ?